docs/error-codes.md

```diff
@@ -0,0 +1,60 @@
+<!--
+- SPDX-FileCopyrightText: 2021 TQ Tezos
+
+- SPDX-License-Identifier: LicenseRef-MIT-TQ
+-->
+
+<!--
+NOTE: This file should not be modified directly.
+Use `stack scripts/generate_error_code.hs` instead.
+-->
+
+## Error Codes
```

Comment:
Documenting it in markdown is nice, however the issue specifically asks for `TZIP-16` error codes, which should be included in the metadata, see the `"errors"` in the [reserved fields](https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#reserved-fields) section.

I don't see this added in this PR.

Reply:
I missed the `errors` fields, I've added this now, thanks Pasquale.

Also, we probably want #305 to be merged first, since I rebase this on top of it due to some errors are removed.

Reply:
@pasqu4le Since #305 is probably not merged any time soon, I rebase this PR on top of master instead and add error code for the removed errors for now. Hope to get your re-review on this.

Reply:
@pasqu4le I've rebased this on the latest master, and fix the CI, let me know if this is good to be merged.